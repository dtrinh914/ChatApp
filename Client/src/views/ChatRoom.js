import React, {useState} from 'react';
import io from 'socket.io-client';
import ChatWindow from '../components/ChatWindow';
import ChatInput from '../components/ChatInput';
import './ChatRoom.css';

let socket = io('http://localhost:3000');
function ChatRoom(){
    const [messages, setMessages] = useState([]);

    //Sends message to backend
    function newMessage(message){
        socket.emit('chat message', message);
        setMessages([...messages, message])
    }

    //listens for new messages from the backend and updates state
    socket.on('chat message', (message) => {
        setMessages([...messages, message])
    });

    
    return(
        <div>
            <ChatWindow messages={messages} />
            <ChatInput newMessage={newMessage} />
        </div>
    );
}

export default ChatRoom;