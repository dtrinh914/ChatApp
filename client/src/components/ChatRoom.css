.ChatRoom{
    height: 95vh;
    width:100%;
}