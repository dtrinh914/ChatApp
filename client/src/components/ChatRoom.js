import React, {useEffect, useContext} from 'react';
import ChatWindow from './ChatWindow';
import ChatInput from './ChatInput';
import './ChatRoom.css';
import io from 'socket.io-client';
import {ChatContext} from '../contexts/chatContext'

let socket;
function ChatRoom(){
    const {chatData, dispatch} = useContext(ChatContext);
    
    useEffect(() => {
        socket = io();
        //listens for new messages from the backend and updates state
        socket.on('message', (room, message) => {
            dispatch({type:'NEWMSG', room:room, message:message})
        });
        
        //on connect joins the rooms on the client side
        socket.on('connect', () => {
            chatData.groups.forEach(group => {
                socket.emit('room', group._id)
            })
        })
    },[])

    const newMessage = (message) => {
        socket.emit('message', chatData.selected, message);
        dispatch({type:'NEW_MSG', room:chatData.selected, message:message});
    }

    return(
        <div className='ChatRoom'>
            <ChatWindow />
            <ChatInput onConfirm={newMessage} />
        </div>
    );
}

export default ChatRoom;