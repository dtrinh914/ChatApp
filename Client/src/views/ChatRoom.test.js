import React from 'react';
import ChatRoom from './ChatRoom';
import { render, cleanup, fireEvent } from '@testing-library/react';

afterEach(cleanup);

test('should match snapshot', () => {
    const {asFragment} = render(<ChatRoom />)
    expect(asFragment()).toMatchSnapshot();
});

test('should type Hello World in chat input, press send, and Hello World should be displayed in chat window', ()=> {
    const {container} = render(<ChatRoom />);
    const ChatInputBox = container.querySelector('.ChatInput textarea');
    const ChatInputButton = container.querySelector('.ChatInput button');
    fireEvent.change(ChatInputBox, {target: {value: 'Hello World'}});
    fireEvent.click(ChatInputButton);

    const message = container.querySelector('.ChatWindow li').textContent;
    expect(message).toBe('Hello World');
})