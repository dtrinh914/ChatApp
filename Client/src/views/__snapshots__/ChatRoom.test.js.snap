// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="ChatWindow"
    >
      <ul />
    </div>
    <form
      class="ChatInput"
    >
      <input
        type="text"
        value=""
      />
      <button>
        Send
      </button>
    </form>
  </div>
</DocumentFragment>
`;
